dali/c_api_2/pipeline.cc

```diff
@@ -0,0 +1,302 @@
+// Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/c_api_2/pipeline.h"
+#include "dali/c_api_2/pipeline_outputs.h"
+#include "dali/c_api_2/error_handling.h"
+#include "dali/pipeline/pipeline.h"
+#include "dali/c_api_2/utils.h"
+#include "dali/c_api_2/validation.h"
+
+namespace dali::c_api {
+
+PipelineWrapper *ToPointer(daliPipeline_h handle) {
+  if (!handle)
+    throw NullHandle("Pipeline");
+  return static_cast<PipelineWrapper *>(handle);
+}
+
+PipelineWrapper::PipelineWrapper(const daliPipelineParams_t &params) {
+  bool pipelined = true, async = true, dynamic = false, set_affinity = false;
+  if (params.exec_type_present) {
+    pipelined = params.exec_type & DALI_EXEC_IS_PIPELINED;
+    async = params.exec_type & DALI_EXEC_IS_ASYNC;
+    dynamic = params.exec_type & DALI_EXEC_IS_DYNAMIC;
+  }
+  if (params.exec_flags_present) {
+    set_affinity = params.exec_flags & DALI_EXEC_FLAGS_SET_AFFINITY;
+  }
+  pipeline_ = std::make_unique<Pipeline>(
+    GET_OR_DEFAULT(params, max_batch_size, -1),
+    GET_OR_DEFAULT(params, num_threads, -1),
+    GET_OR_DEFAULT(params, device_id, -1),
+    GET_OR_DEFAULT(params, seed, -1_i64),
+    pipelined,
+    GET_OR_DEFAULT(params, prefetch_queue_depth, 2),
+    async,
+    dynamic,
+    0,
+    set_affinity);
+}
+
+PipelineWrapper::PipelineWrapper(
+      const void *serialized,
+      size_t length,
+      const daliPipelineParams_t &params) {
+  bool pipelined = true, async = true, dynamic = false, set_affinity = false;
+  if (params.exec_type_present) {
+    pipelined = params.exec_type & DALI_EXEC_IS_PIPELINED;
+    async = params.exec_type & DALI_EXEC_IS_ASYNC;
+    dynamic = params.exec_type & DALI_EXEC_IS_DYNAMIC;
+  }
+  if (params.exec_flags_present) {
+    set_affinity = params.exec_flags & DALI_EXEC_FLAGS_SET_AFFINITY;
+  }
+  pipeline_ = std::make_unique<Pipeline>(
+    std::string(static_cast<const char *>(serialized), length),
+    GET_OR_DEFAULT(params, max_batch_size, -1),
+    GET_OR_DEFAULT(params, num_threads, -1),
+    GET_OR_DEFAULT(params, device_id, -1),
+    pipelined,
+    GET_OR_DEFAULT(params, prefetch_queue_depth, 2),
```

Comment:
This means we always ignore the `prefetch_queue_depth` argument from pipeline definition, right?

Reply:
That's a bug.

Reply:
I don't think it's serialized.